server.go

```diff
@@ -32,39 +36,126 @@ const (
 	runtime        = "Microsoft AzureKMS"
 	runtimeVersion = "0.0.1"
 	configFilePath  = "/etc/kubernetes/azure.json"
+	azureResourceURL = "https://management.azure.com/"
+	vaultResourceURL = "https://vault.azure.net"
 )
 
+// KMSServiceServer is a gRPC server.
 type KMSServiceServer struct {
 	pathToUnixSocket string
 	net.Listener
 	*grpc.Server
 }
 
+// New creates an instance of the KMS Service Server.
 func New(pathToUnixSocketFile string) *KMSServiceServer {
 	kmsServiceServer := new(KMSServiceServer)
 	kmsServiceServer.pathToUnixSocket = pathToUnixSocketFile
 	fmt.Println(kmsServiceServer.pathToUnixSocket)
 	return kmsServiceServer
 }
 
-func getKeysClient() (kv.ManagementClient, string, string, string, error)  {
-	vmClient := kv.New()
-	token, err := GetKeyvaultToken(AuthGrantType(), configFilePath)
+func getKey() (kv.ManagementClient, string, string, string, error)  {
+	kvClient := kv.New()
+
+	vaultURL, err := getVault()
 	if err != nil {
-		return vmClient, "", "", "", fmt.Errorf("failed to get token, error: %v", err)
+		return kvClient, "", "", "", fmt.Errorf("failed to get vault, error: %v", err)
 	}
 	
-	vmClient.Authorizer = token
-	vaultBaseURL, keyName, keyVersion, err := GetKMSProvider(configFilePath)
+	token, err := GetKeyvaultToken(AuthGrantType(), configFilePath, vaultResourceURL)
 	if err != nil {
-		return vmClient, "", "", "", fmt.Errorf("failed to get KMS provider configs, error: %v", err)
+		return kvClient, "", "", "", fmt.Errorf("failed to get token, error: %v", err)
+	}
+	
+	kvClient.Authorizer = token
+	_, keyName, keyVersion, err := GetKMSProvider(configFilePath)
+	if keyName == nil {
+		return kvClient, "", "", "", fmt.Errorf("failed to find KMS providerKeyName in config")
+	}
+	if keyVersion == nil {
+		fmt.Println("Unable to find KMS providerKeyVersion in configs, getting keyVersion")
+		keyVersion, err := getKeyVersion(kvClient, *vaultURL, *keyName)
```

Comment:
the keyvault client treats empty `""` keyvault version as `latest`, I think we should assume empty key version on configuration == `stay on latest`